src/avram/associations/belongs_to.cr

```diff
@@ -48,6 +48,24 @@ module Avram::Associations::BelongsTo
 
   private macro define_belongs_to_base_query(assoc_name, model, foreign_key)
     class BaseQuery
+      def self.preload_{{ assoc_name }}(record, preload_query = {{ model }}::BaseQuery.new)
+        preload_{{ assoc_name }}(records: [record], preload_query: preload_query).first
+      end
+
+      def self.preload_{{ assoc_name }}(records : Enumerable, preload_query = {{ model }}::BaseQuery.new)
```

Comment:
Should this modify the passed in records or return new records? Right now the passed in records are being mutated, but I'm not sure if we have a way of getting an exact copy.

Reply:
It seems to me mutating records would be more performant.  In general Avam seems to have moved to duping database interaction for safety.  I think updating the `records` themselves might mitigate some unexpected behavior.

Reply:
here is an example of non mutating refactor.

```crystal
results = {} of {{ model }}::PrimaryKeyType => Array({{ model }})
ids = records.compact_map(&.{{ foreign_key }})

results if ids.empty?
{{ assoc_name }} = preload_query.dup.id.in(ids).results.group_by(&.id)

records.each do |record|
  if (id = record.{{ foreign_key }})
    rec = record.dup
    rec.__set_preloaded_{{ assoc_name }} {{ assoc_name }}[id]?.try(&.first?)
    results << rec
  else
    rec = record.dup
    rec.__set_preloaded_{{ assoc_name }} nil
    results << rec
  end
end

results
```

Reply:
Unfortunately I had to split the implementations between the class level and the instance level because the instance level relies on the mutations but I decided to go down this route. Thanks for looking into this 👍